chapters/edo/main.tex

```diff
@@ -1,2 +1,1193 @@
+
+\begin{figure}[htbp]
+    \centering
+    \begin{subfigure}{\imgwidth}
+        \includegraphics[width=\imgwidth]{kmeans_silhouette_coverage.pdf}
+        \caption{%
+            with the silhouette fitness function%
+        }\label{fig:silhouette_coverage}
+    \end{subfigure}
+
+    \vspace{1em}
+    \begin{subfigure}{\imgwidth}
+        \includegraphics[width=\imgwidth]{kmeans_discounted_coverage.pdf}
+        \caption{with the discounted silhouette fitness function}
+    \end{subfigure}
+    \caption{%
+        Scatter and density plots of the selected parents at 10 epoch
+        intervals from the examples in Section~\ref{subsec:silhouette}
+    }\label{fig:coverage}
+\end{figure}
+
+Although this does provide evidence to say that the current design of the EA can
+sufficiently explore its given search space with an appropriate fitness
+function, it does not provide any guarantee that this will happen, even in
+expectation, with any fitness function.
+
+Another weakness of EAs is their tendency to find the `easy' way out. That is,
+reducing down to the most straightforward solution which solves the given
+problem. In most cases, that is not a problem and is often, in fact, favourable.
+The concentrated diagonal in Figure~\ref{fig:silhouette_coverage} shows this
+sort of reductive behaviour. In that particular example, the most natural
+solution for the EA (i.e.\ to maximise the silhouette coefficient with
+\(k\)-means) was to attempt to collapse one dimension of the search space to
+make the problem one-dimensional. This kind of behaviour is not necessarily a
+bad thing as trivial, basic and straightforward cases are of great importance
+when understanding an algorithm's quality.
+
+However, should that be a problem, then the objective function could be adjusted
+accordingly. The case study in this chapter examined several iterations of
+fitness functions, but each was adjusted according to what was apparent at the
+time. These adjustments were possible because of the architecture of the
+implementation of this method. A similar strategy could be employed
+automatically by a more sophisticated fitness function that retains some
+information about the datasets generated from previous runs of EDO on a
+particular (or at least similar) parameter set. In this way, the currently
+completely unsupervised learning conducted by the EA could be ushered away from
+less helpful solutions (via some penalty, say) and toward previously unexplored
+behaviours. This automatic, iterative application of the proposed method would
+likely reveal more sophisticated insights into a particular algorithm.
+
+In essence, the EDO method is merely a tool that demonstrates the benefit of the
+flipped paradigm set out in this work. The concept of where `good' datasets
+exist is not well-documented in the literature, and this thesis hopes that
+Evolutionary Dataset Optimisation acts as a starting point for further works to
+come.
```

Comment:
As mentioned above, somewhere in this section we need to point at the matching chapter.

Reply:
Fixed by 024815b